### DISCUSSION TITLE
About the SocketKeyStore

### DISCUSSION THREAD

**Discussion by wybdev03** [quickfix-j/quickfixj#679] [Q&A]
SocketKeyStore=DEMO.p12
SocketKeyStorePassword=Demo

The SocketKeyStore parameter currently specifies a file path. Does it support code acquisition of certificate information and setting it in the configuration

**Comment by wybdev03** [comment]
![image](https://github.com/quickfix-j/quickfixj/assets/142007940/838ed4c2-a89d-4a9f-8012-4d968563b35b)
Does it support obtaining keyStore through non reading files？

**Comment by the-thing** [comment]
Not at the moment. `quickfix.SessionSettings` only store key/value information which later is decided what to do with them. `quickfix.mina.ssl.SSLContextFactory` assumes that the value of `SocketKeyStore` is the file name.

A good approach perhaps could be to expose an interface for `quickfix.mina.ssl.SSLContextFactory` so you could provide your own ssl context factory.

**Reply by wybdev03** [reply]
Do you have a plan on how to expose an interface?

**Reply by the-thing** [reply]
Currently this is not possible and requires a change in QuickFIXJ.

The change would require socket acceptor/initiatior to allow to set a custom SSLContextFactory instance. I imagine this would look similar to the way you pass a custom MessageFactory instance.

**Comment by the-thing** [comment]
@wybdev03

BTW. This is a question so this should go in the discussions first.

**Reply by chrjohn** [reply]
Thanks @the-thing , converted.
